### DISCUSSION TITLE
How to ignore duplicate errors, continue to insert ones that are not duplicates

### DISCUSSION THREAD

**Discussion by kaykhan** [prisma/prisma#14192] [Q&A] [answered]
Im fetcing some data and i want to insert documents into mongodb. Some of those documents will be duplicates based on a uniqueness key i have defined, and others will not be duplicates. 

When there is a duplicate in the batch of documents, it fails. I want it to insert what it can. 

In mongodb i would use unordered inserts for this [ `.insertMany(payload, { ordered: false })` see reference link below ]. How do i achieve this in prisma client.

```
model InsightHistory {
    id            String    @id @default(auto()) @map("_id") @db.ObjectId
    address       String
    network       String
    date          DateTime?
    avg_price     String?
    max_price     String?
    min_price     String?
    trades        Int?
    unique_buyers Int?
    volume        String?

    @@unique([address, date], name: "uniq_address_date")
    @@map("insightHistory")
}
```

**attempt 1:**

```
        const r = await PrismaService.insightHistory.createMany({ data: payload });
```

> Error: Unique constraint failed on the constraint: `insightHistory_address_date_key`\

**attempt 2**

```
        const response = await PrismaService.$runCommandRaw({
            insert: "insightHistory",
            documents: payload as any,
        });
```   

> "n":0,"ok":1,"writeErrors":[{"code":11000,"errmsg":"E11000 duplicate key error collection: nft-rarify.insightHistory index: insightHistory_address_date_key dup key

I was hoping i can get it to insert what it can and ignore these duplicate errors

-- https://stackoverflow.com/questions/61480444/mongodb-insertmany-and-skip-duplicates

**Comment by TasinIshmam** [comment, marked as answer]
Hi @kaykhan  

Can you try modifying the `runCommandRaw` as follows: 

```
  const response = await PrismaService.$runCommandRaw({
            insert: "insightHistory",
            documents: payload as any,
            ordered : false
        });
```

You would also need to wrap the method in a try-catch block because, as the [StackOverflow question you referenced](https://stackoverflow.com/questions/61480444/mongodb-insertmany-and-skip-duplicates), this will still raise an error (but the insertion should succeed).

**Reply by kaykhan** [reply]
@TasinIshmam okay thats strange, this was one of the first iterations of $runCommandRaw i tried and originally it did not work. But after trying it again based on your suggestions, it works now. :/

Also it does not get caught here in the try catch like it traditionally would using the mongo node driver.  The response looks something looks like...

`n` being the number of documents that have been inserted

```
  n: 3,
  writeErrors: [
    {
      index: 0,
      code: 11000,
      keyPattern: [Object],
      keyValue: [Object],
      errmsg: 'E11000 duplicate key error collection: nft-rarify.insightHistory index: insightHistory_address_date_key dup key: { address: "0x495f947276749ce646f68ac8c248420045cb7b5e", date: "2020-12-06T00:00:00Z" }'
    }...],
```

thanks!

**Reply by TasinIshmam** [reply]
Alright awesome, happy to help :D

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
